File: Problems/136.single-number.py
# title: Single Number
# timestamp: 2025-05-20 11:12:13
# problemUrl: https://leetcode.com/problems/single-number/
# memory: 19.9 MB
# runtime: 7 ms

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        counter = Counter(nums)
        for n, i in counter.items():
            if i == 1:
                return n
        
        return -1

File: Problems/136.single-number.go
// title: Single Number
// timestamp: 2024-04-18 13:22:00
// problemUrl: https://leetcode.com/problems/single-number/
// memory: 6.3 MB
// runtime: 13 ms

func singleNumber(nums []int) int {
    var res int
    for _, num := range nums {
        res ^= num
    }
    return res
}